Python/dp/Knapsack0-1.py

```diff
@@ -0,0 +1,69 @@
+"""
+Knapsack 0-1 problem using dp (0-1 means we either choose it or we don't, no fractions)
+Given weights and their corresponding values,
+We fill knapsack of capacity W to obtain maximum possible value in bottom-up manner.
+
+N: Number of (items)weight elements 
+W: Capacity of knapsack
+Time Complexity: O(N*W).
+Space Complexity: O(N*W). 
+
+
+"""
+
+
+def knapsack(capacity, weights, values):
+    # items= number of items
+    items = len(weights)
+
+    # Initializing knapsack_table values as 0
+    knapsack_table = [[0 for x in range(capacity + 1)]
+                     for x in range(items + 1)]
+
+    # knapsack_table[][] in bottom up manner
+    for i in range(items + 1):
+
+        # Note that list is 0-based, so to get ith item we do i-1
+        for j in range(capacity + 1):
+
+            # i=0: Zero items to put
+            # j=0: zero capacity left
+            if i == 0 or j == 0:
+                knapsack_table[i][j] = 0
+
+            # weight of item is greater than current capacity, so we cannot use it
+            # so knapsack value here is the best we can do without this item ie. with the (i-1)th item
+            elif weights[i-1] > j:
+                knapsack_table[i][j] = knapsack_table[i-1][j]
+
+            # weight of item is less than capacity, so can use it to optimize knapsack
+            else:
+                knapsack_table[i][j] = max(
+                    # if ith item is included,
```

Comment:
Can you try making the Comments more consistent as per grammar (That's not a necessity but it would look good 😅 )